### PR TITLE
[FLINK-19420][docs-zh] Translate "Program Packaging" page into Chinese

### PR DESCRIPTION
**Pull request by Shawn-Hx**
## What is the purpose of the change

Translate "Program Packaging" page into Chinese.

The page url is https://ci.apache.org/projects/flink/flink-docs-master/zh/dev/packaging.html

The markdown file is located in flink/docs/dev/packaging.zh.md

## Brief change log

Translate "Program Packaging" page into Chinese.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by Shawn-Hx** [comment]
Hi, @klion26 
This PR is ready to be reviewed. Could you help to review it at your convenience?
Thanks.

File: docs/dev/packaging.zh.md [hunk 1]
```diff
@@ -24,36 +24,20 @@ under the License.
...
-automatically when exporting JAR files.
+正如之前所描述的，Flink 程序可以使用 `remote environment` 在集群上执行。或者，程序可以被打包成 JAR 文件（Java Archives）执行。如果使用[命令行]({% link ops/cli.zh.md %})的方式执行程序，将程序打包是必需的。
+
+### 打包程序
+
+为了能够通过命令行或 web 界面执行打包的 JAR 文件，程序必须使用通过 `StreamExecutionEnvironment.getExecutionEnvironment()` 获取的 environment。当 JAR 被提交到命令行或 web 界面后，该 environment 会扮演集群环境的角色。如果调用 Flink 程序的方式与上述接口不同，该 environment 会扮演本地环境的角色。
+
+打包程序只要简单地将所有相关的类导出为 JAR 文件，JAR 文件的 manifest 必须指向包含程序*入口点*（拥有公共 `main` 方法）的类。实现的最简单的方法是将 *main-class* 写入 manifest 中（比如 `main-class: org.apache.flinkexample.MyProgram`）。*main-class* 属性与 Java 虚拟机通过指令 `java -jar pathToTheJarFile` 执行 JAR 文件时寻找 main 方法的类是相同的。大多数 IDE 提供了在导出 JAR 文件时自动包含该属性的功能。
 
 ### Summary
```
Comment (klion26) [inline docs/dev/packaging.zh.md:35]: 这个标题也翻译下？

File: docs/dev/packaging.zh.md [hunk 2]
```diff
@@ -24,36 +24,20 @@ under the License.
...
-To package the program, simply export all involved classes as a JAR file. The JAR file's manifest
-must point to the class that contains the program's *entry point* (the class with the public
-`main` method). The simplest way to do this is by putting the *main-class* entry into the
-manifest (such as `main-class: org.apache.flinkexample.MyProgram`). The *main-class* attribute is
-the same one that is used by the Java Virtual Machine to find the main method when executing a JAR
-files through the command `java -jar pathToTheJarFile`. Most IDEs offer to include that attribute
-automatically when exporting JAR files.
+正如之前所描述的，Flink 程序可以使用 `remote environment` 在集群上执行。或者，程序可以被打包成 JAR 文件（Java Archives）执行。如果使用[命令行]({% link ops/cli.zh.md %})的方式执行程序，将程序打包是必需的。
+
+### 打包程序
```
Comment (klion26) [inline docs/dev/packaging.zh.md:29]: 标题翻译之后，需要添加 `<a>` 标签，可以参考 [wiki](https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications) 锚点可以参考 英文版的 url

File: docs/dev/packaging.zh.md [hunk 3]
```diff
@@ -24,36 +24,20 @@ under the License.
...
+
+为了能够通过命令行或 web 界面执行打包的 JAR 文件，程序必须使用通过 `StreamExecutionEnvironment.getExecutionEnvironment()` 获取的 environment。当 JAR 被提交到命令行或 web 界面后，该 environment 会扮演集群环境的角色。如果调用 Flink 程序的方式与上述接口不同，该 environment 会扮演本地环境的角色。
+
+打包程序只要简单地将所有相关的类导出为 JAR 文件，JAR 文件的 manifest 必须指向包含程序*入口点*（拥有公共 `main` 方法）的类。实现的最简单的方法是将 *main-class* 写入 manifest 中（比如 `main-class: org.apache.flinkexample.MyProgram`）。*main-class* 属性与 Java 虚拟机通过指令 `java -jar pathToTheJarFile` 执行 JAR 文件时寻找 main 方法的类是相同的。大多数 IDE 提供了在导出 JAR 文件时自动包含该属性的功能。
```
Comment (klion26) [inline docs/dev/packaging.zh.md:33]: `实现的最简单的方法` -> `实现的最简单方法` 是否会更好一些呢

**Review by klion26 — COMMENTED** [review]
@Shawn-Hx  thanks for the contribution, left some minor comments. please take a look.

**Comment by Shawn-Hx** [comment]
Thanks for your kind reminder.
I have made the changes. PTAL.

**Review by klion26 — APPROVED** [review]
@Shawn-Hx thanks for the update. LGTM now
cc @wuchong @rmetzger

**Pull request merged**